### PR TITLE
Menu Section: VoiceOver improvements Mark II

### PR DESCRIPTION
**Pull request by etoledom**
Fixes second part of https://github.com/wordpress-mobile/WordPress-iOS/issues/12881

This PR adds some extra accessibility labels and hints to make this section more friendly with VoiceOver.
The section design itself makes it difficult to make a better implementation with VoiceOver, since there are many elements appearing and disappearing, elements with many actions including swappable views.

At the end, and after testing a few different things, I have decided to go by the simple way and add labels to missing elements.

**To test:**

- Run WPiOS on this branch.
- Navigate to the Menus section.
- Turn on [VoiceOver ](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/docs/using-voiceover.md)
- Check that there are no elements without a proper accessibility description.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m
Comment (jd-alexander) [inline WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m]:
```suggestion
    textField.accessibilityLabel = NSLocalizedString(@"Menu name", @"Screen Reader: Description for text field that edits the menu name.");
```

**Comment by jkmassel** [comment]
👋 We're freezing `15.8` today, so I'm bumping this PR forward to `15.9`. If you need it to be part of the `15.8` release cycle, please merge to the `release/15.8` branch and DM me – I'll be happy to cut a new beta release!

**Comment by jd-alexander** [comment]
@etoledom I haven't reviewed this as yet because I was having issues with my development certificate that would allow WPiOS to run on my iPhone. I just got it working so I will do a review tomorrow 🙇

**Review by jd-alexander — APPROVED** [review]
Thanks for these changes @etoledom Tested using VoiceOver on an iPhone 11 and works great! LGTM 🚢 I was wondering if this branch needs to be updated with `develop` because when I was trying to run it I was having some build failures so just checking to be sure. Once all is well then you can merge.

**Comment by etoledom** [comment]
Thank you for the review @jd-alexander !

Were you building with Xcode 12?
To be sure, I'll update the branch as you suggested and merge after ✅ from CI.

**Comment by jd-alexander** [comment]
@etoledom oh yes I was building with XCode 12 :)

**Pull request merged**
